api/token.go

```diff
@@ -324,11 +336,22 @@ func (a *API) IdTokenGrant(ctx context.Context, w http.ResponseWriter, r *http.R
 		return badRequestError("Could not read id token grant params: %v", err)
 	}
 
-	if params.IdToken == "" || params.Nonce == "" || params.Provider == "" {
-		return oauthError("invalid request", "id_token, nonce and provider required")
+	if params.IdToken == "" || params.Nonce == "" {
+		return oauthError("invalid request", "id_token and nonce required")
+	}
+	
+	if params.Provider == "" && ( params.ClientID == "" || params.Issuer == "" ) {
+		return oauthError("invalid request", "provider or client_id and issuer required")
```

Comment:
shouldn't we need either the `Provider` or both the `ClientID` and `Issuer` ?

Reply:
Clarification.

case 00  
  params.Provider == ""  
  params.ClientID == ""  
  params.Issuer == ""  
-> error

case 01  
params.Provider == ""  
params.ClientID == "abcde"  
params.Issuer == ""  
-> error

case 02  
params.Provider == ""  
params.ClientID == ""  
params.Issuer == "https://abcde"  
-> error

case 03  
params.Provider == "abcde"  
params.ClientID == ""  
params.Issuer == ""  
-> not error

case 04  
params.Provider == ""  
params.ClientID == "abcde". 
params.Issuer == "https://abcde"  
-> not error

case 05  
params.Provider == "abcde"  
params.ClientID == "abcde"  
params.Issuer == "https://abcde"  
-> not error

Reply:
There is no need to change this part of the code, is there?

Reply:
oh ok sry i misread it, yeah there's no need to change this